homeassistant/components/demo/__init__.py

```diff
@@ -224,76 +228,82 @@ async def demo_start_listener(_event):
-                "mean": mean,
-                "min": mean - random() * max_diff,
-                "max": mean + random() * max_diff,
-            }
+            StatisticData(
+                start=now,
+                mean=mean,
+                min=mean - random() * max_diff,
+                max=mean + random() * max_diff,
+            )
         )
         now = now + datetime.timedelta(hours=1)
 
     return statistics
 
 
-def _generate_sum_statistics(start, end, init_value, max_diff):
-    statistics = []
+def _generate_sum_statistics(
+    start: datetime.datetime, end: datetime.datetime, init_value: float, max_diff: int
+) -> list[StatisticData]:
+    statistics: list[StatisticData] = []
     now = start
     sum_ = init_value
     while now < end:
         sum_ = sum_ + random() * max_diff
         statistics.append(
-            {
-                "start": now,
-                "sum": sum_,
-            }
+            StatisticData(
+                start=now,
+                sum=sum_,
+            )
         )
         now = now + datetime.timedelta(hours=1)
 
     return statistics
 
 
-async def _insert_statistics(hass):
+async def _insert_statistics(hass: HomeAssistant) -> None:
     """Insert some fake statistics."""
     now = dt_util.now()
     yesterday = now - datetime.timedelta(days=1)
     yesterday_midnight = yesterday.replace(hour=0, minute=0, second=0, microsecond=0)
 
     # Fake yesterday's temperatures
-    metadata = {
-        "source": DOMAIN,
-        "statistic_id": f"{DOMAIN}:temperature_outdoor",
-        "unit_of_measurement": "°C",
-        "has_mean": True,
-        "has_sum": False,
-    }
+    metadata = StatisticMetaData(
+        source=DOMAIN,
+        statistic_id=f"{DOMAIN}:temperature_outdoor",
+        unit_of_measurement="°C",
+        has_mean=True,
+        has_sum=False,
+        name=None,
+    )
     statistics = _generate_mean_statistics(
         yesterday_midnight, yesterday_midnight + datetime.timedelta(days=1), 15, 1
     )
     async_add_external_statistics(hass, metadata, statistics)
 
     # Fake yesterday's energy consumption
-    metadata = {
-        "source": DOMAIN,
-        "statistic_id": f"{DOMAIN}:energy_consumption",
-        "unit_of_measurement": "kWh",
-        "has_mean": False,
-        "has_sum": True,
-    }
+    metadata = StatisticMetaData(
+        source=DOMAIN,
+        statistic_id=f"{DOMAIN}:energy_consumption",
+        unit_of_measurement="kWh",
+        has_mean=False,
+        has_sum=True,
+        name=None,
+    )
     statistic_id = f"{DOMAIN}:energy_consumption"
     sum_ = 0
     last_stats = await get_instance(hass).async_add_executor_job(
         get_last_statistics, hass, 1, statistic_id, True
     )
-    if "domain:energy_consumption" in last_stats:
-        sum_ = last_stats["domain.electricity_total"]["sum"] or 0
+    if f"{DOMAIN}:energy_consumption" in last_stats:
+        sum_ = last_stats[f"{DOMAIN}.electricity_total"]["sum"] or 0
```

Comment:
@emontnemery this fails mypy validation, and I am not sure how to fix.
I think this piece of code is simply never called, and maybe those two lines can simply be removed, but since you added them I think it's better to check with you...

Reply:
I have adjusted this area based on #77206

Reply:
Solved following rebase